section2/reflection.md

```diff
@@ -2,14 +2,37 @@
 
-5. How do you add multiple conditions to an `if` statement? In your own words, explain how to program reads them and determines what to do.
+A conditional statement is a block of code that will return an output that depends on certain conditions that can be modified using variables.  A daily life conditional would be something like a height requirement to ride a roller coaster at a theme park.  _If_ you are `variable` inches tall, you may ride the roller coaster.  _Else_ you may not ride the roller coaster.  A web application that makes use of a conditional is the homepage for [4 Noses Brewing Company](https://www.4nosesbrewing.com/). When you first click that link, a box comes up asking if you are 21 years of age or older.  If you click yes, you enter the website, and if you click no, you are re-directed to the website for Capri-Sun.
+
+5. How do you add multiple conditions to an `if` statement? In your own words, explain how the program reads them and determines what to do.
+
+```javascript
+if (variable === x) {
+  console.log("This will be printed");
+} else if (variable === y) {
+  console.log("That will be printed");
+} else {
+  console.log("These will be printed");
+}
+```
+In the above example, the program will first check if the `variable` is equal to x. If it is, it will print the first result.  If it does not equal x, it will move on to the next line.  If the `variable` is equal to y, it will print the second result.  If neither of the first two `if` statements are true, it will move to the third and final result and print it because I have not set a specific condition for that result.  So if the `variable` is anything besides x or y, it will print the third result.
 
 6. What tools are available to you, as a developer, if you want to check that TWO conditions are met? Or, if you want to check that 1 of 2 conditions are met, before running a block of code?
 
-7. What questions do you still have about `if` statements and/or functions?
\ No newline at end of file
+You can add multiple conditions to a single line of an `if` statement by using a logical operator.  These are: and `&&`, or `||`.  When using `&&` the program will require that both statements surrounding the `&&` are true in order to run that line of code.  When using `||` the program only requires that one of the two statements are true.
+
+7. What questions do you still have about `if` statements and/or functions?
+
+I am feeling really good about `if` statements so far.  One question that I do have is about the semi-colons after every `console.log` line. It doesn't seem like these are necessary, as the code runs without error with or without them.  Is this just a "best practice" type of thing? I could understand that with more complicated coding it could help to read and visualize the line breaks, but for something more simple like the exercises we've done so far, is it really necessary?
```

Comment:
Yes, such a great question. There are some HOT debates about this out there on the internet. Most FE devs and teams will use the semi-colons to explicitly communicate to the browser that the line is over. That's why you don't see the `;` after a line that starts and `if` statement.

Are they really necessary? In most cases, no; just used for best practice/consistency/team norming. There are rare cases where omitting a semi-color WILL cause a problem and it can be extremely difficult to hunt down as JS is not known to provide the most helpful errors. 

Turing, along with probably any company you'll work for as a dev, has a style guide that advises on it - including using semi-colons at the end of a line of JS.

Reply:
(the company you work for might not call for semi-colons - I meant that they will likely have a style guide!)